dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/navigation/NavResultHydrated.java

```diff
@@ -0,0 +1,183 @@
+            if (req != null) {
+                // We exclude the page name from the Request URI so we can check if page's parent
+                // object is the real active object
+                String reqURI = req.getRequestURI();
+                String parentPath = reqURI.substring(0, reqURI.lastIndexOf("/"));
+                if (!parentPath.endsWith("/"))
+                    // Adding a slash at the end of the path, so it avoids false positives
+                    // when two or more paths from the same level starts with the same name
+                    parentPath = parentPath + "/";
+                // If the current item is a folder, we check if it's part of current URI
+
+                // System.err.println(href + " : " + reqURI);
+                if (isFolder() && !navResult.getHref()
+                    .endsWith("/")) {
+                    String tempHref = navResult.getHref() + "/";
+                    return parentPath.startsWith(tempHref);
+                } else {
+                    // If it's a page, we check instead if it's the current URI
+                    return !isCodeLink() && navResult.getHref()
+                        .equalsIgnoreCase(reqURI);
+                }
+            }
+        }
+        return false;
+    }
+    @Override
+    public String getCodeLink() {
+        if (navResult.getCodeLink() != null && (navResult.getCodeLink()
+            .contains("$")
+                || navResult.getCodeLink()
+                    .contains("#"))) {
+            return UtilMethods.evaluateVelocity(navResult.getCodeLink(), context.getVelocityContext());
+        } else {
+            return navResult.getCodeLink();
+        }
+    }
+    @Override
+    public String getTitle() throws Exception {
+        return navResult.getTitle();
+    }
+
+    @Override
+    public String getHostId() {
+        return navResult.getHostId();
+    }
+
+
+    public String getHref() {
+        return navResult.getHref();
+    }
+
+    @Override
+    public boolean isShowOnMenu() {
+        return navResult.isShowOnMenu();
+    }
+
+    @Override
+    public boolean isCodeLink() {
+        return navResult.isCodeLink();
+    }
+
+    @Override
+    public int getOrder() {
+        return navResult.getOrder();
+    }
+
+
+    @Override
+    public boolean isFolder() {
+        return navResult.isFolder();
+    }
+
+    @Override
+    public List<? extends NavResult> getChildren() throws Exception {
+
+
+        List<NavResultHydrated> list = navResult.getChildren().stream().map(result -> new NavResultHydrated(result, this.context)).collect(Collectors.toList());
+        
+        
+        
+        if (Config.getBooleanProperty("ENABLE_NAV_PERMISSION_CHECK", false)) {
+            // now filtering permissions
+            List<NavResult> allow = new ArrayList<NavResult>(list.size());
+
+            HttpServletRequest req = (HttpServletRequest) context.getRequest();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unnecessary cast to "HttpServletRequest". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1905)